### PR TITLE
Allows users to exclude DefaultMismatchedInputException

### PR DESCRIPTION
**Pull request by geoand**
A few users have run into an issue where
they are trying to handle all Jackson exceptions
but the existence of `DefaultMismatchedInputException`
prevents that for `MismatchedInputException`
(as per JAX-RS / Jakarta REST rules).

This change allows users to do use:

```properties
quarkus.class-loading.removed-resources."io.quarkus\:quarkus-rest-jackson"=io/quarkus/resteasy/reactive/jackson/runtime/mappers/DefaultMismatchedInputException.class
```

in order to have Quarkus ignore the mapper.

_P.S. We could also introduce a specific property for this, but currently I am on the fence about that..._

### CODE REVIEW

**Review by gsmet — COMMENTED** [review]
Hmmm, I have a few comments here.

Should we rename the class to `DefaultMismatchedInputExceptionMapper` before documenting it somewhere? When I read the description, I thought it was an exception and had a hard time understanding the issue.

Second question: isn't it easier to create a mapper with a higher priority that specifically handles this exception instead of promoting a rather obscure feature?

And finally, I wonder if at some point, these dev modes features should be handled specifically after the exception mappers have been applied (and only be triggered if no ExceptionMapper handled the exception).

**Comment by geoand** [comment]
> Should we rename the class to DefaultMismatchedInputExceptionMapper before documenting it somewhere? When I read the description, I thought it was an exception and had a hard time understanding the issue

How about `BuiltinMismatchedInputExceptionMapper`?

> Second question: isn't it easier to create a mapper with a higher priority that specifically handles this exception instead of promoting a rather obscure feature?

That doesn't work (as per JAX-RS / Jakarta REST) when users want to handle a super class of  `MismatchedInputException`.

**Comment by gsmet** [comment]
> How about BuiltinMismatchedInputExceptionMapper?

Looks good.

> That doesn't work (as per JAX-RS / Jakarta REST) when users want to handle a super class of MismatchedInputException.

Maybe I wasn't clear but my understanding was that they would have to have a specific mapper for `MismatchedInputException`. I thought it might be easier to grasp from a user perspective.

But you can start your way and see how to make it easier to grasp later.

> I don't really understand this, can you elaborate a little?

I think it's not the first time we have dev mode features in the way of how you would expect a standard JAX-RS implementation to work.
Typically this feature we have is nice in dev mode but I think it shouldn't be in the way when you customize things on your side.

Thus why I was wondering if we should have a way to enable these features only if something from the user didn't catch things earlier.

Basically, we should find a way for our mappers (or not implement them as mappers) to be very last resort if nothing else from the users intervened first.

**Comment by geoand** [comment]
> Maybe I wasn't clear but my understanding was that they would have to have a specific mapper for MismatchedInputException. I thought it might be easier to grasp from a user perspective.

That would work today without any configuration. What does not work as users would like, is handling all exceptions that are super classes of this exception. 

> Basically, we should find a way for our mappers (or not implement them as mappers) to be very last resort if nothing else from the users intervened first.

Yeah, that would be nice, but unfortunately the resolution rules don't really allow this - hence the ability to remove this one.

**Comment by geoand** [comment]
PR updated

File: docs/src/main/asciidoc/rest.adoc [hunk 1]
```diff
@@ -1415,9 +1415,28 @@ In both cases, importing those modules will allow HTTP message bodies to be read
...
+====
+There are situations where various Jackson related exceptions need to handled in a uniform way.For example, the application may need to handle all `JsonMappingException` the same way.
+This becomes a problem when taking JAX-RS / Jakarta REST rules into account, because the exception mapper `ExceptionMapper` for `MismatchedInputException` would be used instead of the user provide
+`ExceptionMapper` for `JsonMappingException` (as `MismatchedInputException` is a subtype of `JsonMappingException`).
+
+One solution for this case is to configure the following:
+
+[source,properties]
+----
+quarkus.class-loading.removed-resources."io.quarkus\:quarkus-rest-jackson"=io/quarkus/resteasy/reactive/jackson/runtime/mappers/BuiltinMismatchedInputException.class
```
Comment (gsmet) [inline docs/src/main/asciidoc/rest.adoc:1435]:
```suggestion
quarkus.class-loading.removed-resources."io.quarkus\:quarkus-rest-jackson"=io/quarkus/resteasy/reactive/jackson/runtime/mappers/BuiltinMismatchedInputExceptionMapper.class
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I will handle the comment given I'm already on this subject, no need for you to context switch.

**Comment by gsmet** [comment]
Force pushed with the doc fix squashed!

**Comment by geoand** [comment]
Thanks!

**Review by gsmet — APPROVED** [review]
